sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1207,6 +1207,17 @@ class Analyzer(
                 val alias = Alias(ae, ae.toString)()
                 aggregateExpressions += alias
                 alias.toAttribute
+              // Replacing [[NamedExpression]] causes the error on [[Grouping]] because the
+              // grouping column will be new attribute created by adding additional [[Alias]].
+              // So we can't find the grouping column and replace it in the rule
+              // [[ResolveGroupingAnalytics]].
```

Comment:
what about we do some improve for next case:

1. add one more check: the expression should not exist in the chid output
2. if the expression is a `NamedExpression` already, don't alias it.